casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -25,7 +25,7 @@ object Estimator {
     */
   def tips[F[_]: Monad: BlockStore](
       blockDag: BlockDag,
-      genesisBlockHash: BlockHash
+      lastFinalizedBlock: BlockMessage
```

Comment:
I don't think you need to pass in the whole `BlockMessage` here. I think passing in the `BlockHash` as before is sufficient and you can look up the block number (which is the only other field you use as far as I can tell) from `blockDag.dataLookup`.